### PR TITLE
[deckhouse] New config backend. Use ModuleConfig instead of Configmap

### PR DESCRIPTION
**Pull request by yalosev**
## Description
Set custom backend for kube-config-manager, which work based on ModuleConfig instead of ConfigMap
Also boostrap process handles ModuleConfig instead of `configOverrides`
`pointer.` lib deprecated functions were refactored. Hail the linter

## Why do we need it, and what problem does it solve?

We used to use configmap to configure deckhouse and move the data from ModuleConfigs there. This created an unnecessary layer of abstraction that led to errors.
Now we can use ModuleConfig as the primary source and read the configuration directly from it.
Also ModuleConfig is now used when bootstrap clusters, instead of configOverrides

## What is the expected result?
Also
Close [#5828](https://github.com/deckhouse/deckhouse/issues/5828)

## Checklist
- [x] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: deckhouse-controller
type: feature
summary: Use ModuleConfig as primary source of configuration. We don't use configmap `deckhouse` anymore.
impact_level: default
```

```changes
section: dhctl
type: feature
summary: Use ModuleConfig to override the default configuration instead of `configOverrides` section.
impact_level: default
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by yalosev** [comment]
/e2e/abort yandex-cloud;WithoutNAT;containerd;1.25 6499248050 6499248050-1-con-1-25 /sys/deckhouse-oss/install:pr6061 cloud-user@84.201.173.85

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've started the workflow for label 'e2e/run/aws'...

**Comment by yalosev** [comment]
/e2e/abort aws;WithoutNAT;containerd;1.25 6546928580 6546928580-1-con-1-25 /sys/deckhouse-oss/install:pr6061 centos@35.156.202.108

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by yalosev** [comment]
/e2e/abort aws;WithoutNAT;containerd;1.25 6549155501 6549155501-1-con-1-25 /sys/deckhouse-oss/install:pr6061 centos@3.66.14.1

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by yalosev** [comment]
/e2e/abort yandex-cloud;WithoutNAT;containerd;1.25 6573347233 6573347233-1-con-1-25 /sys/deckhouse-oss/install:pr6061 cloud-user@158.160.125.37

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by yalosev** [comment]
/e2e/abort yandex-cloud;WithoutNAT;containerd;1.25 6576338606 6576338606-1-con-1-25 /sys/deckhouse-oss/install:pr6061 cloud-user@158.160.61.93

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by yalosev** [comment]
/e2e/abort yandex-cloud;WithoutNAT;containerd;1.25 6578297571 6578297571-1-con-1-25 /sys/deckhouse-oss/install:pr6061 cloud-user@51.250.83.127

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by yalosev** [comment]
/e2e/abort yandex-cloud;WithoutNAT;containerd;1.25 6579907842 6579907842-1-con-1-25 /sys/deckhouse-oss/install:pr6061 cloud-user@84.201.131.232

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by yalosev** [comment]
/e2e/abort yandex-cloud;WithoutNAT;containerd;1.25 6580173089 6580173089-1-con-1-25 /sys/deckhouse-oss/install:pr6061 cloud-user@51.250.78.151

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've started the workflow for label 'e2e/run/openstack'...

**Comment by z9r5** [comment]
/e2e/abort openstack;Standard;containerd;1.25 6580874931 6580874931-1-con-1-25 /sys/deckhouse-oss/install:pr6061 debian@31.129.33.89

**Comment by z9r5** [comment]
/e2e/abort yandex-cloud;WithoutNAT;containerd;1.25 6580628935 6580628935-1-con-1-25 /sys/deckhouse-oss/install:pr6061 cloud-user@84.201.128.13

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by yalosev** [comment]
/e2e/abort yandex-cloud;WithoutNAT;containerd;1.25 6580628935 6580628935-1-con-1-25 /sys/deckhouse-oss/install:pr6061 cloud-user@84.201.128.13

File: deckhouse-controller/pkg/addon-operator/kube-config/backend/module_config.go [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+			mconfig := obj.(*v1alpha1.ModuleConfig)
+			mc.handleEvent(mconfig, eventC)
+		},
+		DeleteFunc: func(obj interface{}) {
+			mc.handleEvent(obj.(*v1alpha1.ModuleConfig), eventC)
+		},
+	})
+	if err != nil {
+		// TODO: return err
+		panic(err)
```
Comment (yalosev) [inline deckhouse-controller/pkg/addon-operator/kube-config/backend/module_config.go:75]: Look here

Comment (yalosev) [reply deckhouse-controller/pkg/addon-operator/kube-config/backend/module_config.go]: Fixed

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've started the workflow for label 'e2e/run/static'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've started the workflow for label 'e2e/run/gcp'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've started the workflow for label 'e2e/run/eks'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've started the workflow for label 'e2e/run/openstack'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've started the workflow for label 'e2e/run/azure'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've started the workflow for label 'e2e/run/vsphere'...

**Comment by yalosev** [comment]
/e2e/abort gcp;WithoutNAT;containerd;1.25 6659762786 6659762786-1-con-1-25 /sys/deckhouse-oss/install:pr6061

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by yalosev** [comment]
/e2e/abort azure;Standard;containerd;1.25 6659762827 6659762827-1-con-1-25 /sys/deckhouse-oss/install:pr6061 azureuser@20.224.20.151

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've started the workflow for label 'e2e/run/gcp'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've started the workflow for label 'e2e/run/azure'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've started the workflow for label 'e2e/run/gcp'...

**Review by RomanenkoDenys — APPROVED** [review]
LGTM

**Review by name212 — DISMISSED** [review]

**Pull request merged**